### PR TITLE
[DOCS] Security best practices for GitHub Actions workflows development

### PR DESCRIPTION
**Pull request by ababushk**
### Details:
Security guidelines for GitHub Actions workflows developers

### CODE REVIEW

File: docs/dev/ci/github_actions/security.md [hunk 1]
```diff
@@ -0,0 +1,24 @@
+# Security best practices for GitHub Actions Workflows
+
+There are a few simple steps that we should follow to ensure our workflows are not vulnerable to common attacks:
+
+- Use the “permissions” key to make sure the GITHUB_TOKEN is configured with the least privileges for each job. Check [GitHub documentation](https://docs.github.com/en/actions/writing-workflows/choosing-what-your-workflow-does/controlling-permissions-for-github_token) on this also.
```
Comment (mryzhov) [inline docs/dev/ci/github_actions/security.md:5]: I would suggest to provide default value, e.g:
`permissions: read-all`

**Comment by akladiev** [comment]
Could you add the link to this doc section to the entrypoint page?https://github.com/ababushk/openvino/blob/security_best_practices_doc/docs/dev/ci/github_actions/overview.md

File: docs/dev/ci/github_actions/security.md [hunk 2]
```diff
@@ -0,0 +1,24 @@
...
+
+- Declare environment variables at the step level when possible
+
+- Never use `pull_request_target` trigger event for workflows. If you want to use `pull_request_target`, contact a member of OpenVINO GitHub Actions task force first. Check [GitHub blog post](https://securitylab.github.com/resources/github-actions-preventing-pwn-requests/) on this as well.
+
+- Never use plain-text secrets hard-coded in GitHub Actions Workflow. If you need to use secrets, contact a member of OpenVINO GitHub Actions task force first.
```
Comment (akashchi) [inline docs/dev/ci/github_actions/security.md:11]:
```suggestion
- Never use plain-text secrets hard-coded in GitHub Actions Workflow. If you need to use secrets, contact a member of the OpenVINO GitHub Actions task force first.
```

File: docs/dev/ci/github_actions/security.md [hunk 3]
```diff
@@ -0,0 +1,24 @@
+# Security best practices for GitHub Actions Workflows
+
+There are a few simple steps that we should follow to ensure our workflows are not vulnerable to common attacks:
+
+- Use the “permissions” key to make sure the GITHUB_TOKEN is configured with the least privileges for each job. Check [GitHub documentation](https://docs.github.com/en/actions/writing-workflows/choosing-what-your-workflow-does/controlling-permissions-for-github_token) on this also.
+
+- Declare environment variables at the step level when possible
```
Comment (mryzhov) [inline docs/dev/ci/github_actions/security.md:7]: As opposed to the global environment or when no environment is used at all?

File: docs/dev/ci/github_actions/security.md [hunk 4]
```diff
@@ -0,0 +1,24 @@
...
+
+- Never use `pull_request_target` trigger event for workflows. If you want to use `pull_request_target`, contact a member of OpenVINO GitHub Actions task force first. Check [GitHub blog post](https://securitylab.github.com/resources/github-actions-preventing-pwn-requests/) on this as well.
```
Comment (akashchi) [inline docs/dev/ci/github_actions/security.md:9]:
```suggestion
- Never use `pull_request_target` trigger event for workflows. If you want to use `pull_request_target`, contact a member of the OpenVINO GitHub Actions task force first. Check [GitHub blog post](https://securitylab.github.com/resources/github-actions-preventing-pwn-requests/) on this as well.
```

File: docs/dev/ci/github_actions/security.md [hunk 5]
```diff
@@ -0,0 +1,24 @@
...
+
+- Be careful with user input.
+
+    Most of GitHub context variables propagated from user input. That means they should be treated as an untrusted and potentially malitious. There are some tactics you can use to mitigate the risk:
+    - Instead of using inline scripts, create an action and pass the variable as an argument
+    - Put the value into an environment variable for the step, and use the variable in the script
+
+    More details are available in [this](https://securitylab.github.com/resources/github-actions-untrusted-input/) blog post.
+
+- When using third-party actions, pin the version with a commit hash rather than a tag to shield your workflow from potential supply-chain compromise.
```
Comment (mryzhov) [inline docs/dev/ci/github_actions/security.md:21]: Let's insert an example:
`uses: actions/checkout@11bd71901bbe5b1630ceea73d27597364c9af683 # v4.2.2`

**Review by mryzhov — APPROVED** [review]

File: docs/dev/ci/github_actions/security.md [hunk 6]
```diff
@@ -0,0 +1,24 @@
...
+
+- Declare environment variables at the step level when possible
+
+- Never use `pull_request_target` trigger event for workflows. If you want to use `pull_request_target`, contact a member of OpenVINO GitHub Actions task force first. Check [GitHub blog post](https://securitylab.github.com/resources/github-actions-preventing-pwn-requests/) on this as well.
+
+- Never use plain-text secrets hard-coded in GitHub Actions Workflow. If you need to use secrets, contact a member of OpenVINO GitHub Actions task force first.
+
+- Be careful with user input.
+
+    Most of GitHub context variables propagated from user input. That means they should be treated as an untrusted and potentially malitious. There are some tactics you can use to mitigate the risk:
```
Comment (akashchi) [inline docs/dev/ci/github_actions/security.md:15]:
```suggestion
    Most GitHub context variables are propagated from user input. That means they should be treated as untrusted and potentially malicious. There are some tactics you can use to mitigate the risk:
```

File: docs/dev/ci/github_actions/security.md (hunk 1 above)
Comment (akashchi) [inline docs/dev/ci/github_actions/security.md:5]:
```suggestion
- Use the `permissions` key to make sure the `GITHUB_TOKEN` is configured with the least privileges for each job. Check [GitHub documentation](https://docs.github.com/en/actions/writing-workflows/choosing-what-your-workflow-does/controlling-permissions-for-github_token) on this also.
```
I would also suggest adding the example from either our workflows or from the GH docs.

File: docs/dev/ci/github_actions/security.md (hunk 5 above)
Comment (akashchi) [inline docs/dev/ci/github_actions/security.md:21]: I would suggest adding an example from out workflows.

**Review by akladiev — APPROVED** [review]

**Review by mryzhov — APPROVED** [review]

File: docs/dev/ci/github_actions/security.md (hunk 6 above)
Comment (ababushk) [reply docs/dev/ci/github_actions/security.md]: Oops 😅 thanks a lot!

File: docs/dev/ci/github_actions/security.md (hunk 1 above)
Comment (ababushk) [reply docs/dev/ci/github_actions/security.md]: Added

**Pull request merged**